File: KotlinLibraryX/app-api/src/main/java/com/lee/api/adapter/ViewBindingItem.java
package com.lee.api.adapter;

import androidx.viewbinding.ViewBinding;

/**
 * @author jv.lee
 * @date 2019/3/29
 * 某一类的item对象
 */
public interface ViewBindingItem<T> {

    /**
     * 获取item的布局
     *
     * @return 布局id
     */
    ViewBinding bindViewBinding();

    /**
     * 是否开启item点击
     *
     * @return boolean值
     */
    boolean openClick();

    /**
     * 是否打开防抖
     *
     * @return boolean
     */
    boolean openShake();

    /**
     * 是否打开回收回调
     * @return boolean
     */
    boolean openRecycler();

    /**
     * 是否为当前的item布局
     *
     * @param entity
     * @param position
     * @return
     */
    boolean isItemView(T entity, int position);

    /**
     * 将item的控件与需要显示数据绑定
     *
     * @param holder
     * @param entity
     * @param position
     */
    void convert(ViewBindingHolder holder, T entity, int position);

    /**
     * 当item隐藏后 回调销毁及回收操作
     * @param holder
     * @param entity
     * @param position
     */
    void viewRecycled(ViewBindingHolder holder, T entity, int position);

}

File: KotlinLibraryX/library/src/main/java/com/lee/library/adapter/item/ViewBindingItem.kt
package com.lee.library.adapter.item

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import com.lee.library.adapter.base.BaseViewHolder
import com.lee.library.adapter.base.BaseViewItem
import com.lee.library.adapter.binding.ViewBindingHolder
import com.lee.library.tools.ViewBindingTools

/**
 * viewBinding实现 item类型类
 * @author jv.lee
 * @date 2021/6/15
 */
abstract class ViewBindingItem<VB : ViewBinding, Data> : BaseViewItem<Data> {

    private var _binding: VB? = null
    val mBinding: VB get() = _binding!!

    override fun getItemViewAny(context: Context, parent: ViewGroup): Any {
        _binding = ViewBindingTools.inflateWithGeneric(
            this,
            LayoutInflater.from(context), parent, false
        )
        return mBinding
    }

    override fun convert(holder: BaseViewHolder, entity: Data, position: Int) {
        this.convert(holder as ViewBindingHolder, entity, position)
    }

    override fun viewRecycled(holder: BaseViewHolder) {
        this.viewRecycled(holder as ViewBindingHolder)
    }

    // Deprecated use ViewBindingUtil.inflateWithGeneric binding
    // impl ItemBinding.inflate(LayoutInflater.from(context), parent, false)
//    abstract fun getItemViewBinding(context: Context, parent: ViewGroup): ViewBinding

    abstract fun convert(holder: ViewBindingHolder, entity: Data, position: Int)

    open fun viewRecycled(holder: ViewBindingHolder) {}
}